src/lib/libwasmfs_fetch.js

```diff
@@ -133,7 +136,9 @@ addToLibrary({
         var chunkSize = fileInfo.chunkSize;
         var firstChunk = (offset / chunkSize) | 0;
         // See comments in getFileRange.
-        var lastChunk = ((offset+length-1) / chunkSize) | 0;
+        var lastChunk = Math.min(
+          ((offset+length-1) / chunkSize),
+          ((fileInfo.size-1) / chunkSize)) | 0;
```

Comment:
Can we truncate offset and/or length to handle this? That is, when the size is small, the effective length is lower.

Reply:
That would also work.  I'll revise my PR, I like that better.